models/models.py

```diff
@@ -0,0 +1,26 @@
+from odoo import api, models, fields, _
+import datetime as dt
+from odoo.exceptions import ValidationError
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+    active = fields.Boolean(default=True)
+    lifespan = fields.Integer(string='Lifespan')
+    USPhoneNumber = fields.Char(string="Phone Number")
+
+    def archive_purchase_order_form_view(self):
+        for rec in self:
+            rec.active = False
+
+    def archive_purchase_order_list_view(self):
+        for rec in self:
+            if rec.state in ('cancel', 'done'):
+                rec.active = False
+            else:
+                raise ValidationError(_('Archive is only used for Locked or Canceled order!'))

```

Comment:
Should check whether `self` contains records whose status is not in Done or Cancelled and raise the error before performing `for` loop

Reply:
Also, you didn't make this method satisfy the condition `Only Purchase Manager can archive Purchase Order(s)` because later it can be called from elsewhere